functions/src/webhooks/transcription.ts

```diff
@@ -0,0 +1,53 @@
+import * as functions from "firebase-functions"
+import { AssemblyAI } from "assemblyai"
+import { db } from "../firebase"
+import { Hearing } from "../events/types"
+import { sha256 } from "js-sha256"
+
+const assembly = new AssemblyAI({
+  apiKey: process.env.ASSEMBLY_API_KEY ? process.env.ASSEMBLY_API_KEY : ""
+})
+
+export const transcription = functions.https.onRequest(async (req, res) => {
+  if (
+    req.headers["X-Maple-Webhook"] &&
+    req.headers["webhook_auth_header_value"]
+  ) {
+    if (req.body.status === "completed") {
+      const transcript = await assembly.transcripts.get(req.body.transcript_id)
+      if (transcript && transcript.webhook_auth) {
+        const maybeEventInDb = await db
+          .collection("events")
+          .where("videoAssemblyId", "==", transcript.id)
+          .get()
+        if (maybeEventInDb.docs.length) {
+          const authenticatedEventsInDb = maybeEventInDb.docs.filter(e => {
+            const hearing = Hearing.check(e)
```

Comment:
Just double-checking - is this going to have the same `.data()` issue we had before, since `e` is a `QueryDocumentSnapshot`?

Reply:
good call, updated.